### DISCUSSION TITLE
[V4.1]Nginx proxy and client IP address

### DISCUSSION THREAD

**Discussion by ymsaadi** [adonisjs/core#2735] [Help]
Hello,
I have an endpoint in my backend where I want to log the user's IP address whenever he try to access it (through my frontend), But I endup getting my nginx proxy address instead of the client IP.

Note that the one that actually makes the request is my front-end app (React app), so when the client clicks on a button in the front app, My react app sends a post request to my Adonisjs Backend, its in there where I want to log the Client IP address.

I did set the `trustProxy: true,` in config/app.js

Also in my nginx config file I have:

```
location / {
   
 proxy_pass http://localhost:3335;

    proxy_http_version 1.1;

    proxy_set_header Connection "upgrade";

    proxy_set_header Host $host;

    proxy_set_header Upgrade $http_upgrade;

    proxy_set_header X-Real-IP $remote_addr;

    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;

  }
```

Can someone help me why I'm not getting the clients IP address when I use request.ip() and request.ips()

Thank you.

As a follow up and something I didn't mention last time, when I send a request from postman directly to the API, I get the actual client's IP address instead of the NGINX server's IP address, so I think the backend's NGINX configuration is correct.

**Comment by thetutlage** [comment]
Just to make sure, is your frontend app making request on `yourdomain.com` or `yourdomain.com:3335`?

**Reply by ymsaadi** [reply]
My full .conf file is like this:

```
server {
  listen 4004 default_server;
  listen [::]:4004 default_server;

  server_name example.com;

  # Our Node.js application
  location / {
    proxy_pass http://localhost:3335;
    proxy_http_version 1.1;
    proxy_set_header Connection "upgrade";
    proxy_set_header Host $host;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $remote_addr;
    #proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  }
}
```

I use mydomain.com:4004 to make request to my backend.

**Reply by thetutlage** [reply]
Then your request is not going via nginx. You are directly hitting your node service.

**Reply by ymsaadi** [reply]
So my front-end should make requests to mydomain.com:3335 ?

**Reply by thetutlage** [reply]
Ahhh scratch that. I thought `4004` is the port of your node app. The request is indeed going via nginx and as per the current config you should get the correct client ip address. 

Can you share how exactly you are accessing the ip address?

**Reply by ymsaadi** [reply]
Yeah of course:

```
    const ips = request.ips();
    const geo = geoip.lookup(ips[0]);
    console.log(ips);
    const view = new View();
    view.publication_id = issue.publication_id;
    view.ip = ips[0];
    view.country = geo.country;
    view.region = geo.region;
    view.city = geo.city;
```

I also tried request.ip()

Also request.ips() only has 1 element which is the front end IP address. That's why I use ips[0] to access it.

**Reply by thetutlage** [reply]
> Also request.ips() only has 1 element which is the front end IP address

By frontend IP address you mean the nginx server IP address or the user network ip address?

**Reply by ymsaadi** [reply]
I mean the nginx server IP address.

**Reply by ymsaadi** [reply]
Can I get some help on this issue ?
As a follow up and something I didn't mention last time, when I send a request from postman directly to the API, I get the actual client's IP address instead of the NGINX server's IP address, so I think the backend's NGINX configuration is correct.
